torch/testing/_internal/common_quantization.py

```diff
@@ -467,10 +467,12 @@ def _group_quantize_tensor(w, n_bit=4, q_group_size=16):
     scales = (max_val - min_val).clamp(min=1e-6) / max_int
     assert torch.isnan(scales).sum() == 0
 
-    zeros = min_val + scales * (2 ** (n_bit - 1))
+    zeros = min_int - min_val.div(scales).round()
+    zeros = torch.clamp(zeros, min_int, max_int)
+    zeros = zeros.to(torch.int8)
     assert torch.isnan(zeros).sum() == 0
```

Comment:
This is also used in tinygemm, should not change this one.

Reply:
Thanks for pointing out, I have moved the codes to `xpu/test_gemm.py`